### DISCUSSION TITLE
farsi(persian) does not verified in the get method

### DISCUSSION THREAD

**Discussion by Alipira** [fastapi/fastapi#13071] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import pandas as pd
import pathlib
from fastapi import FastAPI, HTTPException

service = FastAPI()
current_path = pathlib.Path(__file__).parents[1]
file_path = current_path / './database/db.csv'

@service.get(path='/user/{password}')
async def read_user(password: int):
    try:
        df = pd.read_csv(file_path, encoding='utf-8')
        user_data = df.query('passwords == @password').to_dict(orient='records')
        if not user_data:
            raise HTTPException(status_code=404, detail=f"Error: User {password} not found")
        return user_data
    except FileNotFoundError:
        raise HTTPException(status_code=404, detail=f"Error: File not found at {file_path}")
    except pd.errors.ParserError:
        raise HTTPException(status_code=400, detail="Error: Failed to parse the CSV file")
```

### Description

When I want to search for a user name that is in Farsi (Persian) I get this error"raise HTTPException(status_code=404, detail=f"Error: User {username} not found")", however when I change the input to password I can retrieve the data

### Operating System

Windows

### Operating System Details

wsl2

### FastAPI Version

0.103.2

### Pydantic Version

1.10.12

### Python Version

3.11.5

**Comment by Knighthawk-Leo** [comment, marked as answer]
It doesn't seem to be an issue with FastAPI. You can try fixing it by ensuring the DataFrame is loaded with utf-8 encoding:
df = pd.read_csv(file_path, encoding='utf-8')
If the issue persists, debug the DataFrame content using print(df.head()) to check how the usernames are being loaded. Also, ensure Farsi usernames don’t have extra spaces or characters that could cause mismatches.
